presto-main/src/main/java/com/facebook/presto/sql/analyzer/RefreshMaterializedViewPredicateAnalyzer.java

```diff
@@ -0,0 +1,167 @@
+        }
+
+        @Override
+        protected Void visitExpression(Expression node, Void context)
+        {
+            throw new SemanticException(NOT_SUPPORTED, node, "Only column specifications connected by logical AND are supported in WHERE clause.");
+        }
+
+        @Override
+        protected Void visitLogicalBinaryExpression(LogicalBinaryExpression node, Void context)
+        {
+            if (!LogicalBinaryExpression.Operator.AND.equals(node.getOperator())) {
+                throw new SemanticException(NOT_SUPPORTED, node, "Only logical AND is supported in WHERE clause.");
+            }
+            if (!(node.getLeft() instanceof ComparisonExpression || node.getLeft() instanceof LogicalBinaryExpression)) {
+                throw new SemanticException(NOT_SUPPORTED, node.getLeft(), "Only column specifications connected by logical AND are supported in WHERE clause.");
+            }
+            if (!(node.getRight() instanceof ComparisonExpression || node.getRight() instanceof LogicalBinaryExpression)) {
+                throw new SemanticException(NOT_SUPPORTED, node.getRight(), "Only column specifications connected by logical AND are supported in WHERE clause.");
+            }
+
+            return super.visitLogicalBinaryExpression(node, null);
+        }
+
+        @Override
+        protected Void visitComparisonExpression(ComparisonExpression node, Void context)
+        {
+            if (!(node.getLeft() instanceof Identifier || node.getLeft() instanceof DereferenceExpression)) {
+                throw new SemanticException(NOT_SUPPORTED, node.getLeft(), "Only columns specified on literals are supported in WHERE clause.");
+            }
+            if (!(node.getRight() instanceof Literal)) {
+                throw new SemanticException(NOT_SUPPORTED, node.getRight(), "Only columns specified on literals are supported in WHERE clause.");
+            }
+
+            ResolvedField resolvedField = viewScope.tryResolveField(node.getLeft()).orElseThrow(() -> missingAttributeException(node.getLeft()));
+            String column = resolvedField.getField().getOriginColumnName().orElseThrow(() -> missingAttributeException(node.getLeft()));
+
+            if (!viewDefinition.getValidRefreshColumns().orElse(emptyList()).contains(column)) {
+                throw new SemanticException(NOT_SUPPORTED, node.getLeft(), "Refresh materialized view by column %s is not supported.", node.getLeft().toString());
+            }
+
+            Map<SchemaTableName, String> baseTableColumns = viewDefinition.getColumnMappingsAsMap().get(column);
+            if (baseTableColumns != null) {
+                for (SchemaTableName baseTable : baseTableColumns.keySet()) {
+                    tablePredicatesBuilder.put(
+                            baseTable,
+                            new ComparisonExpression(node.getOperator(), new Identifier(baseTableColumns.get(baseTable)), node.getRight()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.